packages/api/src/providers/fitbit.ts

```diff
@@ -363,63 +370,50 @@ export class Fitbit extends Provider implements OAuth2 {
 
   async postAuth(token: string, user: ConnectedUser): Promise<void> {
     const { access_token: accessToken, user_id: fitbitUserId, scope: scopes } = JSON.parse(token);
-
-    const subscriptionTypes: Record<FitbitScopes, FitbitCollectionTypesWithoutUserRevokedAccess> = {
-      [FitbitScopes.activity]: "activities",
-      [FitbitScopes.nutrition]: "foods",
-      [FitbitScopes.sleep]: "sleep",
-      [FitbitScopes.weight]: "body",
-    };
-
-    const activeSubscriptions = await this.getActiveSubscriptions(accessToken);
-    if (activeSubscriptions)
-      await this.revokeTokenFromOtherUsers(user, fitbitUserId, activeSubscriptions, accessToken);
-
-    // TODO #652: Implement userRevokedAccess
+    await this.revokeTokenFromOtherUsers(user, fitbitUserId);
 
     // Creates new WH subscriptions based on the user's selected scopes
     await Promise.all(
-      Object.entries(subscriptionTypes).map(async ([key, subscriptionType]) => {
+      Object.entries(Fitbit.subscriptionTypes).map(async ([key, subscriptionType]) => {
         if (scopes.includes(key)) {
           const subscriptionId = `${fitbitUserId}-${subscriptionType}`;
           const subscriptionUrl = `${Fitbit.URL}/${Fitbit.API_PATH}/${subscriptionType}/apiSubscriptions/${subscriptionId}.json`;
-          this.createSubscription(subscriptionUrl, accessToken);
+          await this.createSubscription(subscriptionUrl, accessToken);
         }
       })
     );
   }
 
-  // Finds existing users connected to this Fitbit account, revokes their tokens and sends WH updates about the disconnections
-  async revokeTokenFromOtherUsers(
-    currentUser: ConnectedUser,
-    fitbitUserId: string,
-    activeSubscriptions: FitbitWebhookSubscriptions,
-    accessToken: string
-  ): Promise<void> {
+  /**
+   * Finds all users connected to the same Fitbit user, removes their tokens and deletes their WH subscriptions, if there are any.
+   *
+   * @param currentUser current ConnectedUser who is connecting to Fitbit
+   * @param fitbitUserId Fitbit ID
```

Comment:
nit: "Fitbit user ID", could include `, it might be the same for different instances of ConnectedUser`

Reply:
with the current implementation, that should never be the case anymore, since we'll be removing Fitbit from the providerMap of other users upon a new connection to the same Fitbit user

Reply:
I see, but isn't that what this function is trying to prevent? Thus, the comment `Finds all users connected to the same Fitbit user,...`

So, in theory, this can happen but by calling this function we're preventing this on our app, right?

No worries with the wording specifics, just thought it would be useful to read the parameter and understand it connects with the first part of the comment/documentation.